### PR TITLE
Walkthrough tutorial

### PR DESCRIPTION
**Pull request by Dentosal**
This PR adds a tutorial on how to use the debugger. This is still a bit work-in-progress, as writing it seems to lead to implementing some missing features and UX improvements.

~~Currently blocked by #11~~

### CODE REVIEW

File: docs/walkthrough.md [hunk 1]
```diff
@@ -0,0 +1,316 @@
+## An example project
+
+First, we need a project to debug, so create a new project using
+
+```bash
+forc init --script dbg_example && cd dbg_example
```
Comment (Voxelot) [inline docs/walkthrough.md]:
```suggestion
forc new --script dbg_example && cd dbg_example
```

File: docs/walkthrough.md [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+        "witnesses": [],
+        "receipts_root": "0000000000000000000000000000000000000000000000000000000000000000"
+    }
+}
+```
+
+However, the key `script` in should contain the actual bytecode to execute, i.e. the contents of `out/debug/dbg_example.bin` as a JSON array. The following command can be used to generate it:
+
+```bash
+python -c 'print(list(open("out/debug/dbg_example.bin", "rb").read()))'
```
Comment (Voxelot) [inline docs/walkthrough.md]: hmm this didn't quite work for me, I got this escaped version:

```
python -c 'print(list(open("out/debug/dbg_example.bin", "rb").read()))'
['\x90', '\x00', '\x00', '\x04', 'G', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\xcc', ']', '\xfc', '\xc0', '\x01', '\x10', '\xff', '\xf3', '\x00', '\x1a', 'L', 'P', '\x00', '\x91', '\x00', '\x00', '\x18', 'P', 'A', '0', '\x08', ']', 'C', '\xf0', '\x00', '_', 'M', '\x00', '\x01', 'P', 'A', '0', '\x00', ']', 'C', '\xf0', '\x01', '_', 'M', '\x00', '\x00', 'P', 'A', '0', '\x00', ']', 'E', '0', '\x00', ']', 'C', '\xf0', '\x02', '\x16', 'A', '\x14', '\x00', 's', '@', '\x00', '\x14', '\x90', '\x00', '\x00', '"', 'P', 'A', '0', '\x00', 'P', 'A', '0', '\x00', ']', 'E', '0', '\x00', ']', 'C', '\xf0', '\x00', '\x10', 'A', '\x04', '@', '_', 'M', '\x00', '\x00', 'P', 'A', '0', '\x08', 'P', 'A', '0', '\x08', ']', 'E', '0', '\x01', 'P', 'A', '0', '\x00', ']', 'A', '0', '\x00', '\x1b', 'A', '\x14', '\x00', '_', 'M', '\x00', '\x01', '\x90', '\x00', '\x00', '\x0e', 'P', 'A', '0', '\x08', ']', 'A', '0', '\x01', ']', 'G', '\xf0', '\x03', '\x13', 'E', '\x10', '\x00', 's', 'D', '\x00', '(', '\x90', '\x00', '\x00', '*', '3', '@', '\x00', '\x00', '\x90', '\x00', '\x00', '0', ']', 'K', '\xf0', '\x04', 'P', 'E', '0', '\x10', '_', 'M', ' ', '\x02', 'P', 'E', '0', '\x10', ']', 'E', '0', '\x02', '4', '\x00', '\x04', '\x11', ']', 'C', '\xf0', '\x01', '$', '\x00', '\x00', '\x00', 'G', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x01', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x05', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x00', '\x08']
```

File: docs/walkthrough.md [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+
+16 | 17
+---|----
+1  | 1
+2  | 1
+3  | 2
+4  | 6
+5  | 24
+
+From this we can clearly see that the left side, register `16` is the `counter` variable, and register `17` is `result`. Now we the counter equals it the given argument, so the loop terminates. So when we continue, the program ends:
```
Comment (Voxelot) [inline docs/walkthrough.md]:
```suggestion
From this we can clearly see that the left side, register `16` is the `counter` variable, and register `17` is `result`. Now the counter equals the given factorial function argument `5`, and the loop terminates. So when we continue, the program finishes without encountering any more breakpoints:
```

File: docs/walkthrough.md (hunk 2 above)
Comment (Dentosal) [reply docs/walkthrough.md]: Right, Python version 2/3 compatibility issues of course. I'll specify Python 3.

**Review by Voxelot — APPROVED** [review]

**Review by adlerjohn — APPROVED** [review]
utACK

**Pull request merged**